### PR TITLE
Fix multiSelect ChoiceQuestion when answers have spaces

### PR DESCRIPTION
**Pull request by IceMaD**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #32502 
| License       | MIT
| Doc PR        | N/A

Probleme is explained in the issue

### CODE REVIEW

**Comment by Simperfit** [comment]
If this is really something we want to do, it should be deprecated before doing so, because it will break existing command that rely on this.

**Comment by chalasr** [comment]
Any bugfix might break someone's behavior. 
I think that this qualifies as a bug, and that no one reported it because usually people rely on keys for choosing multi-words answers.

This should apply to the 3.4 branch. And trimming should be disabled conditionally as of #31626 (4.4), we missed this part.

**Comment by IceMaD** [comment]
Without the trim on multiple answers, it might break cases where answers don't have spaces but user input have. Exemple :
break

Maybe set all ChoiceQuestion as Trimmable by default on 4.4 for backward compatibility ?

File: src/Symfony/Component/Console/Tests/Question/ChoiceQuestionTest.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+
+class ChoiceQuestionTest extends TestCase
+{
+    /**
+     * @dataProvider selectUseCases
+     */
+    public function testSelectUseCases($isMiltiSelect, $answers, $expected, $message)
```
Comment (chalasr) [inline src/Symfony/Component/Console/Tests/Question/ChoiceQuestionTest.php]: typo `isMiltiSelect`/`isMultiSelect`

File: src/Symfony/Component/Console/Tests/Question/ChoiceQuestionTest.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            'Fourth response',
+        ]);
+
+        $question->setMultiselect($isMiltiSelect);
+
+        foreach ($answers as $answer) {
+            $validator = $question->getValidator();
+            $actual = $validator($answer);
+
+            $this->assertEquals($actual, $expected, sprintf($message, $answer));
```
Comment (chalasr) [inline src/Symfony/Component/Console/Tests/Question/ChoiceQuestionTest.php]: `sprintf()` seems unnecessary

**Review by chalasr — CHANGES_REQUESTED** [review]

**Comment by chalasr** [comment]
@IceMad yes, trimming must stay enabled by default.

Can you please rebase against the 3.4 branch and change the PR base branch accordingly?

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @IceMaD.

**Comment by chalasr** [comment]
And thank you for contributing to Symfony for the first time. Well done, congratz :)

**Pull request merged**